app/models/rent.rb

```diff
@@ -303,34 +281,50 @@ def check_overbook?(rents)
     return @overbook
   end
 
-  # To make sure that dates are in the future and that d_from < d_til
-  # /d.wessman
-  def date_future
-    if self.d_from.present? && self.d_til.present?
-      if (self.d_from > Time.zone.now-10.minutes) && (self.d_til > self.d_from)
-        return true
-      end
+  # Validates d_from is in the future
+  def d_from_future
+    if d_from.present? && d_til.present? && d_from < Time.zone.now
+      errors.add(:d_from, 'måste vara i framtiden.')
+    end
+  end
 
-      if (self.d_from < Time.zone.now)
-        errors.add(:d_from, 'måste vara i framtiden.')
-      end
-      if (self.d_til < self.d_from)
-        errors.add(:d_til, 'måste vara efter startdatumet.')
-      end
-      return false
-    else
-      return true
+  # Validates d_from is before d_til
+  def dates_ascending
+    if d_from.present? && d_til.present? && d_from > d_til
+      errors.add(:d_til, 'måste vara efter startdatumet.')
     end
   end
 
   # To validate the length of the renting
   # /d.wessman
   def duration?
-    if duration > 48 && no_council? == true
+    if duration > 48
       errors.add(:d_from, ', får inte vara längre än 48 h')
     end
   end
 
+  # Custom validations
+  def overlap
+    if Rent.active.date_overlap(d_from, d_til, id).count > 0
+      errors.add(:d_from, 'överlappar med annan bokning')
+    end
+  end
+
+  def overlap_council
+    if Rent.active.date_overlap(d_from, d_til, id).councils.count > 0
+      errors.add(:d_from, 'överlappar med annan utskottsbokning')
+    end
+  end
+
+  def overlap_overbook
```

Comment:
Do not know how to make this shorter.
Any ideas?
The ```@overbook``` variable is set to run the after_create-action which overbooks other rents.